scripts/idl/lint/lint_rules_parser.py

```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/env python
+
+import logging
+import os
+import xml.etree.ElementTree
+
+from dataclasses import dataclass, field
+from typing import List, Optional, Mapping
+from lark import Lark
+from lark.visitors import Transformer, v_args, Discard
+import stringcase
+import traceback
+
+try:
+    from .types import RequiredAttributesRule, AttributeRequirement, ClusterRequirement
+except:
+    import sys
+
+    sys.path.append(os.path.join(os.path.abspath(os.path.dirname(__file__)), "..", ".."))
+    from idl.lint.types import RequiredAttributesRule, AttributeRequirement, ClusterRequirement
+
+
+def parseNumberString(n):
+    if n.startswith('0x'):
+        return int(n[2:], 16)
+    else:
+        return int(n)
+
+
+@dataclass
+class RequiredAttribute:
+    name: str
+    code: int
+
+
+@dataclass
+class DecodedCluster:
+    name: str
+    code: int
+    required_attributes: List[RequiredAttribute]
+
+
+def DecodeClusterFromXml(element: xml.etree.ElementTree.Element):
```

Comment:
Suggest validating from schema line whether you are parsing the XML you think you are. ZAP XML and CSA XML are different.

CSA XML does not yet have any matter stuff in it :(

Reply:
I would do this as a followup if at all.

My expectation is actually that "matter is the future" so all XML files we use are actually matter and must be formatted with matter information.